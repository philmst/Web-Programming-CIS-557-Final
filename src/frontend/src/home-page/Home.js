import React from 'react';
// import { Link } from 'react-router-dom';
import './Home.css';

const clickProfile = (props) => {
  const url = '/profile';
  props.changeState({ link: url });
};

function Home(props) {
  const { changeState } = props;
  console.log('props object is: ', props);

  return (
    <div className="container">

      <div className="header">
        <div className="social-media-title">Social Media</div>
        <div className="profile-picture">
          <div className="img" />
        </div>
        <div className="username">Hi, username!</div>
      </div>

      <div className="top-navbar">
        <div className="home-link">Home</div>
        <div className="profile-link" onClick={() => clickProfile(props)} onKeyDown={() => clickProfile(props)} role="link" tabIndex={0}>Profile</div>
      </div>

      <div className="main-container">

        <div className="side-navbar">
          <button type="submit" onClick={() => changeState({ link: '/updates' })} className="updates">Updates</button>
          <button type="submit" className="events">Events</button>
          <button type="submit" className="groups">Groups</button>
          <button type="submit" className="videos">Videos</button>
          <button type="submit" className="photos">Photos</button>
        </div>

        <div className="main-area">

          <div className="post-area">
            User can input status update here
            <div className="text-area" />
            <div className="post-button" />
          </div>

          <div className="updates-area">
            Status updates will be located here
            <div className="box-1">
              <div className="poster-picture" />
              <div className="update-text" />
              <div className="reply-button" />
            </div>

            <div className="box-2">
              <div className="poster-picture" />
              <div className="update-text" />
              <div className="reply-button" />
            </div>
          </div>
        </div>

        <div className="live-updates">
          Live updates
        </div>

      </div>
    </div>
  );
}

export default Home;
