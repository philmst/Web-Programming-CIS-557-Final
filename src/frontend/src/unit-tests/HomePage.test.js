import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Home from '../home-page/Home';

/**
 * @jest-environment jsdom
 */

const changeState = jest.fn();

describe('Test Home Page UI', () => {
  test('change link called when notification button clicked', async () => {
    render(<Router><Home changeState={changeState} state={{username: 'me'}} /></Router>);
    document.getElementsByClassName('notifications')[0].click();
    expect(changeState).toBeCalled();
    expect(changeState.mock.calls[0][0]).toStrictEqual({link: '/notifications'});
  });

  test('Home page renders correctly', () => {
    const component = renderer.create(<Home changeState={changeState} state={{username: 'me'}} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
