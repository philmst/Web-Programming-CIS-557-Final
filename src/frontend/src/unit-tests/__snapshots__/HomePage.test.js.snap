// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home Page UI Home page renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="header"
  >
    <div
      className="social-media-title"
    >
      Social Media
    </div>
    <div
      className="profile-picture"
    >
      <div
        className="img"
      />
    </div>
    <div
      className="username"
    >
      Hi, username!
    </div>
  </div>
  <div
    className="top-navbar"
  >
    <div
      className="home-link"
    >
      Home
    </div>
    <div
      className="profile-link"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="link"
      tabIndex={0}
    >
      Profile
    </div>
  </div>
  <div
    className="main-container"
  >
    <div
      className="side-navbar"
    >
      <button
        className="notifications"
        onClick={[Function]}
        type="submit"
      >
        Notifications
      </button>
      <button
        className="events"
        type="submit"
      >
        Events
      </button>
      <button
        className="groups"
        onClick={[Function]}
        type="submit"
      >
        Groups
      </button>
      <button
        className="videos"
        type="submit"
      >
        Videos
      </button>
      <button
        className="photos"
        type="submit"
      >
        Photos
      </button>
    </div>
    <div
      className="main-area"
    >
      <div
        className="post-create-area"
      >
        User can create post here
        <div
          className="text-area"
        />
        <div
          className="post-button"
        />
      </div>
      <div
        className="post-area"
      >
        Posts can be viewed here
        <div
          className="box-1"
        >
          <div
            className="poster-picture"
          />
          <div
            className="update-text"
          />
          <div
            className="reply-button"
          />
        </div>
        <div
          className="box-2"
        >
          <div
            className="poster-picture"
          />
          <div
            className="update-text"
          />
          <div
            className="reply-button"
          />
        </div>
      </div>
    </div>
    <div
      className="message-updates"
    >
      Message updates
    </div>
  </div>
</div>
`;
